Description: mainly baseline cleanups and debuild fixes for the new
 controller target (no source code changes).
 .
 zerotier-one (1.4.6-1.0~node1.1) unstable; urgency=medium
 .
   * fix parallel makes and update clean targets for ctlr
Author: Stephen L Arnold <nerdboy@gentoo.org>

---
Forwarded: no
Last-Update: 2019-11-22

--- zerotier-one-1.4.6.orig/make-linux.mk
+++ zerotier-one-1.4.6/make-linux.mk
@@ -314,7 +314,7 @@ zerotier-cli: zerotier-one
 	ln -sf zerotier-one zerotier-cli
 
 libzerotiercore.a:	FORCE
-	make CFLAGS="-O3 -fstack-protector -fPIC" CXXFLAGS="-O3 -std=c++11 -fstack-protector -fPIC" $(CORE_OBJS)
+	$(MAKE) CFLAGS="-O3 -fstack-protector -fPIC" CXXFLAGS="-O3 -std=c++11 -fstack-protector -fPIC" $(CORE_OBJS)
 	ar rcs libzerotiercore.a $(CORE_OBJS)
 	ranlib libzerotiercore.a
 
@@ -336,29 +336,30 @@ src-docs:	FORCE
 
 clean: FORCE
 	rm -rf *.a *.so *.o node/*.o controller/*.o osdep/*.o service/*.o ext/http-parser/*.o ext/miniupnpc/*.o ext/libnatpmp/*.o $(CORE_OBJS) $(ONE_OBJS) zerotier-one zerotier-idtool zerotier-cli zerotier-selftest build-* ZeroTierOneInstaller-* *.deb *.rpm .depend debian/files debian/zerotier-one*.debhelper debian/zerotier-one.substvars debian/*.log debian/zerotier-one doc/node_modules ext/misc/*.o debian/.debhelper debian/debhelper-build-stamp docker/zerotier-one
+	rm -rf debian/zerotier-ctlr*.debhelper debian/zerotier-ctlr.substvars debian/zerotier-ctlr
 
 distclean:	clean
 
 realclean:	distclean
 
 official:	FORCE
-	make -j4 ZT_OFFICIAL=1 all
+	$(MAKE) ZT_OFFICIAL=1 all
 
 docker:	FORCE
 	docker build -f ext/installfiles/linux/zerotier-containerized/Dockerfile -t zerotier-containerized .
 
 controller-node:	FORCE
-	make DEFS+="-DZT_CONTROLLER" one
+	$(MAKE) DEFS+="-DZT_CONTROLLER" one
 
 central-controller:	FORCE
-	make -j4 LDLIBS="-L/usr/pgsql-10/lib/ -lpq -Lext/librabbitmq/centos_x64/lib/ -lrabbitmq" CXXFLAGS="-I/usr/pgsql-10/include -I./ext/librabbitmq/centos_x64/include -fPIC" DEFS="-DZT_CONTROLLER_USE_LIBPQ -DZT_CONTROLLER" ZT_OFFICIAL=1 ZT_USE_X64_ASM_ED25519=1 one
+	$(MAKE) LDLIBS="-L/usr/pgsql-10/lib/ -lpq -Lext/librabbitmq/centos_x64/lib/ -lrabbitmq" CXXFLAGS="-I/usr/pgsql-10/include -I./ext/librabbitmq/centos_x64/include -fPIC" DEFS="-DZT_CONTROLLER_USE_LIBPQ -DZT_CONTROLLER" ZT_OFFICIAL=1 ZT_USE_X64_ASM_ED25519=1 one
 
 central-controller-docker:	central-controller
 	docker build -t docker.zerotier.com/zerotier-central/ztcentral-controller:${TIMESTAMP} -f ext/central-controller-docker/Dockerfile .
 
 debug:	FORCE
-	make ZT_DEBUG=1 one
-	make ZT_DEBUG=1 selftest
+	$(MAKE) ZT_DEBUG=1 one
+	$(MAKE) ZT_DEBUG=1 selftest
 
 # Note: keep the symlinks in /var/lib/zerotier-one to the binaries since these
 # provide backward compatibility with old releases where the binaries actually
@@ -413,6 +414,7 @@ debian:	FORCE
 
 debian-clean: FORCE
 	rm -rf debian/files debian/zerotier-one*.debhelper debian/zerotier-one.substvars debian/*.log debian/zerotier-one debian/.debhelper debian/debhelper-build-stamp
+	rm -rf debian/zerotier-ctlr*.debhelper debian/zerotier-ctlr.substvars debian/zerotier-ctlr
 
 redhat:	FORCE
 	rpmbuild --target `rpm -q bash --qf "%{arch}"` -ba zerotier-one.spec
